src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -162,10 +161,6 @@ internal override TypeWithAnnotations GetIteratorElementType(YieldStatementSynta
                     }
                     elementType = TypeWithAnnotations.Create(CreateErrorType());
                 }
-                else if (asyncInterface && !_methodSymbol.IsAsync)
-                {
-                    Error(elementTypeDiagnostics, ErrorCode.ERR_IteratorMustBeAsync, errorLocation, _methodSymbol, returnType);
```

Comment:
>Error(elementTypeDiagnostics, ErrorCode.ERR_IteratorMustBeAsync, errorLocation, _methodSymbol, returnType); [](start = 20, length = 107)

Can we get into similar trouble with errors reported above?

Reply:
I think so, yes. #39992 is indeed similar. I'll address in this PR. Thanks!